pyccl/core.py

```diff
@@ -953,3 +984,41 @@ def _set_background_from_arrays(self, a_array=None, chi_array=None,
                     or (hoh0_array is None) or (growth_array is None)
                     or (fgrowth_array is None)):
                 raise ValueError("Input arrays not parsed.")
+
+    def _set_linear_power_from_arrays(self, a_array=None, k_array=None,
+                                      pk_array=None):
+        """
+        This function initializes the arrays used for parsing
+        a linear power spectrum from input. Call this function
+        to have the power spectrum be read from input and not
+        computed by CCL.
+
+        a_array (array): an array holding values of the scale factor
+        k_array (array): an array holding values of the wavenumber
+            in units of Mpc^-1).
+        pk_array (array): a 2D array containing the values of the power
+            spectrum at the values of the scale factor and the wavenumber
+            held by `a_array` and `k_array`. The shape of this array must be
+            `[na,nk]`, where `na` is the size of `a_array` and `nk` is the
+            size of `k_array`. This array can be provided in a flattened
+            form as long as the total size matches `nk*na`.
+            Note that, if you pass your own Pk array, you
+            are responsible of making sure that it is sufficiently well
+            sampled (i.e. the resolution of `a_array` and `k_array` is high
+            enough to sample the main features in the power spectrum).
+            For reference, CCL will use bicubic interpolation to evaluate
+            the power spectrum at any intermediate point in k and a.
+        """
+        if self.has_linear_power:
+            raise ValueError("Linear power spectrum has been initialized"
+                             "and cannot be reset.")
+        else:
+            if ((a_array is None) or (k_array is None)
+                    or (pk_array is None)):
+                raise ValueError("Input arrays not parsed.")
```

Comment:
This error messages strikes me as potentially being a bit unclear to some users, can we have something slightly more descriptive like "One or more of the input array for a, k, or pk is not parsed"?

Reply:
Yes, thanks for pointing that out!